### PR TITLE
add support for warming up range of terms

### PR DESCRIPTION
**Pull request by trinity-1686a**
required for quickwit-oss/quickwit#2266

not yet tested. It should be final unless I find issues while implementing \#2266, so I'm keeping this as a draft for now

### CODE REVIEW

**Comment by trinity-1686a** [comment]
I did not find any issues. This should be good to go

File: src/core/inverted_index_reader.rs [hunk 1]
```diff
@@ -206,10 +206,50 @@ impl InvertedIndexReader {
...
+        };
+
+        let mut stream = range_builder.into_stream_async().await?;
+
+        let iter = std::iter::from_fn(move || stream.next().map(|(_k, v)| v.clone()));
+
+        // limit on stream is only an optimization to load less data, the stream may still return
+        // more than limit elements.
+        let mut limit = limit;
+        let iter = iter.take_while(move |_e| match limit {
```
Comment (trinity-1686a) [inline src/core/inverted_index_reader.rs]: maybe this should just be `.take(limit.unwrap_or(usize::MAX))`, even on 32b, limiting a range to "just" 4 billion terms shouldn't be an issue (or rather, there would probably be other problems before)

Comment (fulmicoton) [reply src/core/inverted_index_reader.rs:239]: I'd rather have that yes.

Comment (fulmicoton) [reply src/core/inverted_index_reader.rs]: yeah. let's go with the simpler code

Comment (fulmicoton) [reply src/core/inverted_index_reader.rs]: Can you make the change to use Iterator::take

**Review by fulmicoton — APPROVED** [review]

File: src/core/inverted_index_reader.rs (hunk 1 above)
Comment (trinity-1686a) [reply src/core/inverted_index_reader.rs]: done

**Pull request merged**